crates/astria-conductor/src/conductor.rs

```diff
@@ -235,16 +376,62 @@ impl Conductor {
                     info!("reloading is currently not implemented");
                 }
 
-                res = &mut self.sync_done, if !self.sync_done.is_terminated() => {
+                // Start the sequencer reader
+                res = &mut self.da_sync_done, if !self.da_sync_done.is_terminated() && !self.execution_commit_level.is_firm_only() => {
```

Comment:
This block is very complex and I don't think I understand what is going on here.